core/src/Kokkos_Complex.hpp

```diff
@@ -31,6 +31,20 @@
 
 namespace Kokkos {
 
+// complex_get is a helper for tuple protocol
+// Needs to be declared before complex because complex grants it friendship
+namespace Impl {
+template <size_t I, typename ComplexForwardingRef>
+KOKKOS_FUNCTION constexpr auto&& complex_get(
+    ComplexForwardingRef&& z) noexcept {
+  static_assert(I < 2);
+  if constexpr (I == 0)
+    return static_cast<ComplexForwardingRef&&>(z).re_;
```

Comment:
Any particular reason not to use
```suggestion
    return std::forward<ComplexForwardingRef>(z).re_;
```
(which might be more familiar for perfect forwarding)? Concerns about usability in device code?

Reply:
It was more that I'm not actually forwarding it; rather, I just want to make it the same kind of reference that was passed in.

I know I'm not consistent about this, using `std::move` in other places...

Should I change it?

Reply:
`std::move` wouldn't quite behave as expected, though, see (https://godbolt.org/z/hWq1M13r5), or am I missing something? I'd prefer `std::forward` over `static_assert` here.

Reply:
In the two forms of `get` that return r-value references, I use `std::move(z)` instead of `static_cast<complex<RealType>&&>(z)` / `static_cast<const complex<const RealType>&&>(z)`, even though there is no intent to move, because it doesn't require stating the destination type (as that is deduced), so it is less error-prone.  This is inconsistent with the argument I presented for using `static_cast<ComplexForwardingRef&&>(z)` vs. `std::forward<ComplexForwardingRef>(z)`. 

Anyway, I'll change it to `std::forward<ComplexForwardingRef>(z)`.  What is an extra template instantiation in the grand scheme of things? :-)

Reply:
Done.